### ISSUE TITLE
Lint/OutOfRangeRegexpRef picks wrong regex

### ISSUE THREAD

**Issue by ysbaddaden** [rubocop-hq/rubocop#8708]
## Actual behavior

`Lint/OutOfRangeRegexpRef` seems confused when multiple regular expressions are present on a single line, and picks the _wrong_ Regex, leading to false positives _and_ true negatives.

## Steps to reproduce the problem

In the following example `Lint/OutOfRangeRegexpRef` seems to pick the first Regex, in the `gsub` call (`/\s+/`), instead of the final Regex that is populating `$1` (`/foo(bar)/`):

```ruby
if "foo bar".gsub(/\s+/, "") =~ /foo(bar)/
  p $1
end
```
```
x.rb:2:5: W: Lint/OutOfRangeRegexpRef: Do not use out of range reference for the Regexp.
  p $1
    ^^
```

Rubocop won't complain anymore when the Regex in the `gsub` call is modified to capture, but still won't complain when changing the second regex to not capture:

```ruby
if "foo bar".gsub(/(\s+)/, "") =~ /foobar/
  p $1 # always nil
end
```

## RuboCop version

```
0.90.0 (using Parser 2.7.1.4, rubocop-ast 0.4.0, running on ruby 2.6.6 x86_64-linux-musl)
```

**Comment by ysbaddaden** [comment]
Here is another weird example. This time `Lint/OutOfRangeRegexpRef` will pick the following `/\s{2}/` Regex, despite being a match call on `$1` itself:

```ruby
if "some : line " =~ / : (.+)/
  $1.gsub(/\s{2}/, " ")
end
```
```
x.rb:2:3: W: Lint/OutOfRangeRegexpRef: Do not use out of range reference for the Regexp.
  $1.gsub(/\s{2}/, " ")
  ^^
```
